vars/unitTestPost.groovy

```diff
@@ -80,15 +86,9 @@ def call(Map config = [:]) {
   if (config['valgrind_stash']) {
     def valgrind_pattern = config.get('valgrind_pattern', '*.memcheck.xml')
     stash name: config['valgrind_stash'], includes: valgrind_pattern
-  } else {
-
-    // Need to leave this logic in here for backwards compatibility.
-    // Valgrind results need to stashed and reported in a common stage
-    // After all Valgrind tests are run.
-    valgrindReportPublish ignore_failure: ignore_failure,
-                          valgrind_stashes: []
   }
-  if (!stage_info['with_valgrind']) {
+  def no_record_issues = config.get('no_record_issues', true)
+  if ((!stage_info['with_valgrind'] && no_record_issues) || (stage_info['NLT'] == 1)) {
```

Comment:
Since it's boolean:
```suggestion
  if ((!stage_info['with_valgrind'] && no_record_issues) || (stage_info['NLT'] == True)) {
```

Or perhaps even:

```suggestion
  if ((!stage_info['with_valgrind'] && no_record_issues) || (stage_info['NLT'])) {
```

Not sure if the evaluation there will evaluate the result or that the variable just has a value.

Reply:
I'll try this second one.